google/auth/identity_pool.py

```diff
@@ -219,10 +232,28 @@ def __init__(
             *args,
             **kwargs
         )
-        if not isinstance(credential_source, Mapping):
+        if credential_source is None and subject_token_supplier is None:
+            raise exceptions.InvalidValue(
+                "A valid credential source or a subject token supplier must be provided."
+            )
+        if credential_source is not None and subject_token_supplier is not None:
+            raise exceptions.InvalidValue(
+                "Identity pool credential cannot have both a credential source and a subject token supplier."
+            )
+
+        if subject_token_supplier is not None:
+            self._subject_token_supplier = subject_token_supplier
+            _LOGGER.info(
+                "Subject token supplier is being used. Note that the external account credential does not cache the subject token so caching should be implemented in the supplier."
```

Comment:
Same here, I think this could be a doc string note or warning (Sorry for the misunderstanding).